[sarang_daddy] Feature3 PR

sarangdaddy opened the pull request:
# Draft-PR 내용
- 2주차 학습계획 업데이트
- 1주차 리뷰내용 일부 수정 및 적용

# 2주차 고민사항
- JS를 이용하여 동적으로 HTML 요소 만들기..
- CSS 코드 중복 줄이기

sarangdaddy committed:
자바스크립트 전역변수 제거

- init함수 생성하여 init에 포함시켜주었다.
- 불필요한 변수 제거

sarangdaddy committed:
사이드바 "더 보기" 애니메이션 수정

- 펼쳐지고 접히는 애니메이션으로 수정
- 숨겨진 ul을 sidebar -> ul(상위) 자식으로 수정
- overflow : hidden 추가

sarangdaddy committed:
사이드바 애니메이션 수정

- 오픈시 좌측에서 우측으로 나오듯이 애니메이션 수정
- 클로즈시 좌측으로 다시 들어가도록 애니메이션 수정

sarangdaddy committed:
매직넘버 상수로 대체

sarangdaddy committed:
사이드바 스크롤 보이지 않게 하기

- 대상요소::-webkit-scrollbar -> display:none

sarangdaddy committed:
사이드바 확장 리스트 동적 생성

- HTML에서 요소제거
- JS로 요소 동적 할당

sarangdaddy committed:
sideBarDetail 데이터 객체 생성

- JS 파일들 폴더 정리

sarangdaddy committed:
사이드바 디테일 페이지 레이아웃 생성

- 새로운 <aside>를 생성하여 기존 레이아웃 스타일 그대로 가져옴
- "주 메뉴" 돌아가기 버튼 클릭시 사이드바로 돌아감
- 버튼 이벤트는 상위 ul에 위임하여 구현함

crongro reviewed (commented):
이번주도 학습할 내용들이 많네요. 
하나씩 계획을 점검하며 진행해보세요.

sarangdaddy committed:
사이드바 세부페이지 레이아웃 생성

- 사이드바 리스트 클릭시 세부페이지로 이동
- 세부페이지 flex <-> hidden 클래스 추가 삭제로 호출

sarangdaddy committed:
사이드바 세부페이지 링크 연결하기 완료

- <li> 속성에 data-column 값 부여
- 클릭한 <li>의 data-colunm값과 DATA 키값이 일치하는 세부페이지를 오픈하고자함

sarangdaddy committed:
README 업데이트

- 사이드바 항목과 세부페이지 연결까지 완료

crongro commented on data/sideBarData.js:
```diff
@@ -0,0 +1,355 @@
+    "세발 자전거, 스쿠터 및 왜건",
+    "비디오 게임",
+  ],
+  "2-21": [
```
2-21 이런게 순서대로면 없어도 되지 않을까요? 
그냥 배열로 
[[], [[],[],[],...[]]]

crongro commented on index.html:
```diff
@@ -221,6 +188,9 @@
   ></script>
-  <script src="section-view.js"></script>
-  <script src="sidebar-view.js"></script>
+  <script src="./js/section-view.js"></script>
+  <script src="./js/sidebar-view.js"></script>
+  <script src="./js/sideBarExtention.js"></script>
+  <script src="./data/sideBarData.js"></script>
+  <script src="./js/sideBarDetail.js"></script>
```
나중에 ES Modules 로 만들면 이렇게 나열된 방식이 좀 개선되긴 할거에요.
현업에서는 아직도 이렇게 JS여러개를 나눠서 개발하고 그대로 배포하기도 하긴해요.(좋은 방법은 아니지만)

crongro commented on js/section-view.js:
```diff
@@ -0,0 +1,54 @@
+
+  const dimmDivIndex2 = document.querySelector(".dimmed--zIndex2");
+
+  const SET_TIME_OUT_1000 = 1000;
```
1000 이라고 안해도 될 듯.
전 이런 이름으로 할거 같네요.

const loginUILazyTime = 1000;

crongro commented on js/section-view.js:
```diff
@@ -0,0 +1,54 @@
+  loginSectionDetail.addEventListener("mouseout", hiddenLoginSectionDetail);
+
+  const showLocationSectionDetail = () => {
+    dimmDivIndex2.classList.remove("hidden");
```
이렇게 뭔지 알수 없고 단지 순서만 있는 이름은 약간 별로에요.
1,2,3 이런거 없애고 그 자체의 의미가 있다면 붙여주세요.

물론 2,3 이런 인덱스가 의미를 갖는것이 맞다면 그대로 두고요.

crongro commented on js/sideBarDetail.js:
```diff
@@ -0,0 +1,36 @@
+const sideBarList = document.querySelector(".homeSidebar__subTitle");
+
+const selectListInSidebar = (event) => {
+  // 이벤트 타겟이 근처 li를 찾게 한다.
+  const targetSubtitle = event.target.closest("li");
+  // 클릭된 li의 고유번호 data-columns를 찾는다.
+  let columnNumer = targetSubtitle.dataset.columns;
+  // 클릭된 li의 고유번호 data-columns과 일치하는 DATA key값을 찾는다.
+  if (SIDEBAR_DETAIL[columnNumer]) {
+    // 일치하는 key값을 가진 내용의 세부페이지를 보여준다.
+    detailSideBar.classList.remove("hideDetail");
```
주석이 많은건 좋은데, 모든 코드에 주석이 있어야 하나? 
그건 생각해보세요.

crongro commented on js/sideBarExtention.js:
```diff
@@ -0,0 +1,62 @@
+    "영화 및 TV",
+  ];
+
+  const addList = () => {
```
addList 보다 더 구체적인 함수이름으로~ 
appendXXList ?

crongro commented on js/sideBarExtention.js:
```diff
@@ -0,0 +1,62 @@
+
+  const homeSidePlus = document.querySelector(".hideLists");
+  const homeSideSeeAll = document.querySelector(".homeSidebar__mode--open");
+  const homeSideCloseAll = document.querySelector(".homeSidebar__mode--close");
```
homeSideCloseAll 이름은, 왠지 복수형의 변수같은데..그렇지는 않네요.

crongro reviewed (commented):
수고하셨네요.
리뷰참고하시고요~

ahnlook commented on css/components/side-bar.css:
```diff
@@ -3,8 +3,8 @@
-  width: 320px;
-  height: 100%;
+  width: 310px;
+  height: auto;
```
`auto`와 `100%`의 차이를 설명해주세요!

sarangdaddy replied:
100%는 요소의 부모 요소의 크기를 기반으로 크기를 결정합니다.
auto는 콘텐츠의 너비/높이에 따라 자동으로 크기를 조정합니다.

여기서 height를 지정한 요소의 부모는 body로 고정크기가 아니라 가변적이기 때문에 auto가 더 맞다고 생각했어요.
부모가 고정크기를 가진 요소에도 다르게 적용해보면서 차이점을 더 확인해 봐야겠어요!

ahnlook commented on css/components/side-bar.css:
```diff
@@ -14,6 +14,10 @@
   overflow-y: scroll;
 }
 
+.homeSidebar::-webkit-scrollbar {
```
`-webkit`은 뭔지 궁금해요

sarangdaddy replied:
사실 자료를 보고 그냥 따라 사용했는데..

찾아보니 -webkit-은 웹킷 기반의 브라우저에서만 동작하는 비표준 CSS속성 접두사라고 하네요.
현재는 사파리, 크롬 등에서 공통으로 사용 되고 있다고 합니다. (엣지에서도 다행이 작동이 잘되네요)

확실히 알고 사용해야겠어요! 감사합니다. 릴리 :)

ahnlook replied:
저도 배워갑니다! 감사합니다!

ahnlook commented on css/components/side-bar.css:
```diff
@@ -102,34 +113,34 @@
 
 .showAnimation {
   will-change: transform;
```
이 효과가 있고 없고의 차이는 뭔가요?

sarangdaddy replied:
부라우저에게 요소의 어떤 속성이 변경될 예정이라고 미리 알려주는 속성입니다.
transform과 같이 예측 가능한 속성 변경에 사용하면 애니메이션처리를 최적화 해준다고해요.
하지만 예측이 불가능한 경우나 변경이 너무 자주 발생하는 경우에는 오히려 성능이 떨어질수 있다고 해요.

ahnlook commented on css/components/side-bar.css:
```diff
@@ -102,34 +113,34 @@
 .showAnimation {
   will-change: transform;
-  animation: showSidebarSoft 1s ease-in-out forwards;
+  animation: showSidebarSoft 0.5s ease-out forwards;
```
`0.5s`는 `.5s`로 조금 간단하게 표현할 수 있어요!

sarangdaddy replied:
개인적으로는 0.5s가 가독성이 더 좋다고 생각하는데 .5s의 사용이 더 좋을까요...?

ahnlook commented on index.html:
```diff
@@ -123,96 +123,63 @@
           <span>디지털 콘텐츠 및 기기</span>
         </li>
-        <li class="homeSidebar__subTitle hoverIcon">
+        <li class="homeSidebar__subTitle hoverIcon" data-columns="1-1">
```
`data-colums`는 어떤 용도로 쓰이는지 궁금해요

sarangdaddy replied:
데이터 속성 중에 하나인데 저는 사이드바 리스트 항목들을 각각의 column으로 생각해서 사용했어요.
그런데 크롱께서 columns는 서버에서 정해진 아이템에 대한 고유 id로 사용될 수 있다고 하네요.
서버와의 연동에서도 사용되는 데이타 속성을 같이 학습해두고 사용해야 할것 같아요.

ahnlook reviewed (commented):
저의 손길이 느껴지는 코드가 많이 보이네요? ㅋㅋㅋㅋ (장난)
